src/main/java/org/broadinstitute/barclay/argparser/CommandLineArgumentParser.java

```diff
@@ -844,18 +839,26 @@ private String makeArgumentDescription(final ArgumentDefinition argumentDefiniti
         return sb.toString();
     }
 
-    private void usageForPluginDescriptorArgument(final ArgumentDefinition argDef, final StringBuilder sb) {
-        final CommandLinePluginDescriptor<?> descriptor = (CommandLinePluginDescriptor<?>) argDef.parent;
-        // this argument came from a plugin descriptor; delegate to get the list of allowed values
-        final List<String> allowedValues = new ArrayList<>(descriptor.getAllowedValuesForDescriptorArgument(argDef.getLongName()));
-        if (allowedValues.isEmpty()) {
-            sb.append("Any value allowed");
-        } else {
-            allowedValues.sort(String.CASE_INSENSITIVE_ORDER);
-            sb.append("Possible Values: {");
-            sb.append(String.join(", ", allowedValues));
-            sb.append("}");
+    private void usageForPluginDescriptorArgumentIfApplicable(final ArgumentDefinition argDef, final StringBuilder sb) {
+        if (CommandLineParser.getUnderlyingType(argDef.field).equals(String.class)) {
+            for (CommandLinePluginDescriptor<?> descriptor : pluginDescriptors.values()) {
+                // this argument came from a plugin descriptor; delegate to get the list of allowed values
```

Comment:
This comment is out of place now (we haven't yet concluded that this arg came from a descriptor). Maybe just prefix with "See if this argument came from..."